.github/workflows/graphql.yml

```
@@ -0,0 +1,40 @@
+name: GraphQL API Documentation
+on:
+  push:
+    branches:
+      - 'master'
+    paths:
+      - 'internal/graphql/schema/schema.graphql'
+  pull_request:
+    branches:
+      - 'master'
+    paths:
+      - 'internal/graphql/schema/schema.graphql'
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+          token: ${{ secrets.GH_ACCESS_TOKEN }}
+          ref: 'master'
+
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@473e4d8fe5dd94ee328fdfca9f8c9c7afc9dae5e
+      with:
+        ruby-version: 2.7.1
+        bundler-cache: true
+    - name: graphql-docs
+      run: |
+        cd docs
+        bundle install
+        bundle exec rake graphql:compile_docs
+    - name: Commit changes
+      uses: stefanzweifel/git-auto-commit-action@v4

```

```suggestion
      uses: stefanzweifel/git-auto-commit-action@v4
      if: github.event_name != 'pull_request'
```
This here will ensure that we don't make commits to master when this test runs in PRs